app/models/generic_object.rb

```diff
@@ -200,4 +201,10 @@ def _call_automate(method_name, *args)
     ws = MiqAeEngine.deliver(options)
     ws.root['method_result']
   end
+
+  def remove_go_from_all_related_services
+    ServiceResource.where(:resource_id => id).each do |resource|
```

Comment:
Use `:resource => self`.  The `resource` association is polymorphic, if any other object type happens to share the same ID as this generic object you would delete it.  😱